src/worknet-ext/logger/WorkNetLogger.cs

```diff
@@ -0,0 +1,73 @@
+using System.Diagnostics;
+using Neo;
+using Neo.BlockchainToolkit.Persistence;
+using Neo.Ledger;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.Plugins;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+
+namespace WorkNetExt;
+
+public class WorkNetLogger : Plugin
+{
+    NeoSystem? neoSystem;
```

Comment:
Plugins can override the `Configure` method and call the `GetConfiguration` protected method to retrieve configuration information (in lieu of configuring in code). Here's an [example from the TokenTracker plugin](https://github.com/neo-project/neo-modules/blob/master/src/TokensTracker/TokensTracker.cs#L49): 

```cs
protected override void Configure()
{
    IConfigurationSection config = GetConfiguration();
    _dbPath = config.GetValue("DBPath", "TokensBalanceData");
    _shouldTrackHistory = config.GetValue("TrackHistory", true);
    _maxResults = config.GetValue("MaxResults", 1000u);
    _network = config.GetValue("Network", 860833102u);
    _enabledTrackers = config.GetSection("EnabledTrackers").GetChildren().Select(p => p.Value).ToArray();
}
```

Our sample plugin should demonstrate using Configure/GetConfiguration.

Note, `GetConfiguration` reads from the virtual `ConfigFile` property which (eventually) reads from the static `PluginsDirectory` property. The `PluginsDirectory` property - like the main plugin loaded - is relative to the assembly location. I suspect that worknet compatible plugins will need to overrride `ConfigFile` to read from a location that is relative to the two locations we are loading plugins from.

Reply:
One other thing, [Neo 3.6](https://github.com/neo-project/neo/issues/2844) is likely to ship soon. It might be worth updating the logic of the platform `Plugin` class. In particular, making the platform `Plugin.LoadPlugins` method public + take an optional directory parameter. Then maybe we can also modify the `ConfigFile/RootPath/PluginsDirectory` properties to track the folder the plugin was originally loaded from.